libvirt/domain.go

```diff
@@ -193,3 +195,37 @@ func domainGetIfacesInfo(domain libvirt.Domain, rd *schema.ResourceData) ([]libv
 
 	return interfaces, nil
 }
+
+func qemuAgentInterfacesRefreshFunc(domain libvirt.Domain) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+
+		interfaces, err := domain.ListAllInterfaceAddresses(libvirt.DOMAIN_INTERFACE_ADDRESSES_SRC_LEASE)
+
+		if err != nil {
+			log.Printf("[DEBUG] Qemu-agent error: %s", err)
+			return interfaces, "qemu-agent-wait", err
+		}
+
+		log.Printf("[DEBUG] Interfaces obtained via qemu-agent: %+v", interfaces)
+		return interfaces, "qemu-agent-done", nil
+	}
+}
+
+// Retrieve all the interfaces attached to a domain and their addresses.
+func qemuAgentWaitForInterfacesInfo(domain libvirt.Domain) ([]libvirt.DomainInterface, error) {
+	qemuAgentQuery := &resource.StateChangeConf{
+		Pending:    []string{"qemu-agent-wait"},
+		Target:     []string{"qemu-agent-done"},
+		Refresh:    qemuAgentInterfacesRefreshFunc(domain),
+		MinTimeout: 4 * time.Second,
+		Delay:      4 * time.Second, // Wait this time before starting checks
+		Timeout:    60 * time.Second,
+	}
+
+	interfaces, err := qemuAgentQuery.WaitForState()
+	if err != nil {
+		return []libvirt.DomainInterface{}, fmt.Errorf("Error retrieving interface addresses with Qemu-agent: %s", err)
+	}
+
+	return interfaces.([]libvirt.DomainInterface), nil
+}
```

Comment:
As far I remember we need to wait a litttle because the qemu agent need little to be active on domain. I will check thx

Reply:
yop actually make sense what you write. I think however 2 things.. : 

1) `wait_for_lease` should be more called `wait_for_domain_ips` or something like that..
 ( however is a API breaking change.. )

2) i  will need to have a look at the actual wait_for_lease function because at that time it was not working out of the box for the qemu-agent call.

Reply:
As far as I can remember, we had to wait for the agent because the Domain could be booting up and the agent could have not been started yet...

Reply:
yop exactely.

Imho we can think to refactor deeper the function with `wait_for_lease` function but i would do it in separate pr.

We would need to catch the VirError and also correct some behaviour of the wait for lease function.